scalatest/src/main/scala/io/cucumber/core/options/CucumberSuiteOptionsParser.scala

```diff
@@ -0,0 +1,149 @@
+package io.cucumber.core.options
+
+import io.cucumber.core.exception.CucumberException
+import io.cucumber.core.feature.{FeatureWithLines, GluePath}
+import io.cucumber.core.resource.ClasspathSupport.CLASSPATH_SCHEME_PREFIX
+import io.cucumber.core.snippets.SnippetType
+import io.cucumber.scalatest.CucumberSuiteOptions
+import io.cucumber.tagexpressions.{TagExpressionException, TagExpressionParser}
+
+import java.util.regex.Pattern
+import scala.jdk.OptionConverters.RichOptional
+import scala.util.{Failure, Success, Try}
+
+object CucumberSuiteOptionsParser {
+
+  def unsafeParse[T <: CucumberSuiteOptions](
+      options: T
+  ): RuntimeOptionsBuilder = {
+    parse(options) match {
+      case Success(value) => value
+      case Failure(exception) =>
+        println(
+          "An exception happened while parsing CucumberSuite options. This is likely an issue with Cucumber-Scalatest implementation. Please open an issue on GitHub."
+        )
+        exception.printStackTrace()
+        throw exception
+    }
+  }
+
+  def parse[T <: CucumberSuiteOptions](
+      options: T
+  ): Try[RuntimeOptionsBuilder] = {
+    Try {
+      // TODO verify how getClass behaves with the trait
```

Comment:
So let`s say 

```
trait CucumberSuiteOptions
trait CucumberSuiteOptionSpf extends CucumberSuiteOptions
def parse[T <: CucumberSuiteOptions](options: T with CucumberSuiteOptions): String = {
  val clazz = options.getClass
   clazz.toString
}
parse(new CucumberSuiteOptionSpf{}) // val res3: String = class rs$line$16$$anon$1
```

In Welcome to Scala 3.3.1 (17.0.2, Java OpenJDK 64-Bit Server VM).